File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        l=[[1]]
        for i in range(1,rowIndex+1):
            l1=[1]*(i+1)
            for j in range(1,i):
                l1[j]=l[i-1][j-1]+l[i-1][j]
            l.append(l1)
        return l[rowIndex]

File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<vector<int>>dp(rowIndex+1,vector<int>(rowIndex+1,1));
        int c=1;
        for(int i=1;i<rowIndex+1;i++){
            for(int j=1;j<c;j++){
                dp[i][j]=dp[i-1][j-1]+dp[i-1][j];
            }
            c++;
        }
        return dp[rowIndex];
    }
};